RealImage/testing-challenge-2019
# Shanmugaraj  - pull request for Automation Coding Challenge

Hi,

Submitting my code for Automation Coding Challenge.

Please find the details below on coverage.

Kindly let me know if additional details are required.

Thanks,
Shanmugaraj


List of completed items below:
- designed complete framework using Java TestNG
- included retrieving session ID from Web UI through automation
- customised HTML report with one master report and screenshots
- common methods to handle all 4 http requests
- data drivenÂ using excel
- configured to run using "mvn test" command, hence Jenkins ready


Test included in automation:
- Upload file and validate in account
	covering:
	Upload any file into your account
	Check the uploaded files getting listed in your account
- Approve the shared file from the receiver account
- Reject the shared file from the receiver account
	Both covering:
	Get the list of files in your account
	Create another account (same as Prerequisite) and share files from one account to another account
	Approve/Reject the shared file from the receiver account
- Delete any file from your account
	covering:
	Delete any file from your account
	

Due to time constraint and office work, could not complete
- complete error handling
- just covered happy path
